tests/src/test/java/org/eclipse/hono/tests/amqp/CommandAndControlAmqpIT.java

```diff
@@ -284,6 +288,7 @@ public void testSendOneWayCommandSucceeds(
      */
     @ParameterizedTest(name = IntegrationTestSupport.PARAMETERIZED_TEST_NAME_PATTERN)
     @MethodSource("allCombinations")
+    @AssumeMessagingSystem(type = MessagingType.amqp) // TODO remove when Kafka C&C is implemented!
     public void testSendAsyncCommandsSucceeds(
```

Comment:
One thing to note about running the test with Kafka:
The rate at which commands are sent via Kafka will be much higher than the rate via AMQP.

https://github.com/eclipse/hono/blob/ba371dfffc5ff5a0580b6710fd8d5d6dc5ebbe85/tests/src/test/java/org/eclipse/hono/tests/amqp/CommandAndControlAmqpIT.java#L336-L344

When using the AMQP messaging network, the future above will only return after the Disposition update from the device has been received. In the Kafka case it's only about the ACK from the Kafka cluster.

That could mean that when the AMQP adapter sends the commands to the AMQP test receiver at a relatively high rate, it will not process the credits given back from the AMQP test receiver fast enough. This could result in the AMQP adapter command sender not being able to send commands because of missing credits at some point.
At least that is what I noticed when running the tests locally.
There are 2 ways to fix this:
- set the initial credits (`recv.flow(50)`) to the number of commands to be sent (`totalNoOfCommandsToSend`)
- or impose a delay on the `helper.applicationClient.sendAsyncCommand` invocation, waiting for the command to be received by the test receiver.